From b39d1bc79d3fb5b789860d0339f081b873b65d98 Mon Sep 17 00:00:00 2001
From: Nitzan Lavi <nizan.lavi@gmail.com>
Date: Tue, 22 Mar 2022 17:54:52 +0200
Subject: [PATCH 1/2]  Took socfpga.dtsi from version 5.15  Added de10 nano
 with VIP_FB2 & pio periphery for FPGA's LED, SW and keys

 On branch de10nano
 Changes to be committed:
	modified:   arch/arm/boot/dts/socfpga.dtsi
	new file:   arch/arm/boot/dts/socfpga_cyclone5_de10_nano.dts
---
 arch/arm/boot/dts/socfpga.dtsi                |  22 +-
 .../boot/dts/socfpga_cyclone5_de10_nano.dts   | 198 ++++++++++++++++++
 2 files changed, 218 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm/boot/dts/socfpga_cyclone5_de10_nano.dts

diff --git a/arch/arm/boot/dts/socfpga.dtsi b/arch/arm/boot/dts/socfpga.dtsi
index e40422077438..7c1d6423d7f8 100644
--- a/arch/arm/boot/dts/socfpga.dtsi
+++ b/arch/arm/boot/dts/socfpga.dtsi
@@ -89,7 +89,7 @@
 			};
 		};
 
-		base_fpga_region: base-fpga-region {
+		base_fpga_region {
 			compatible = "fpga-region";
 			fpga-mgr = <&fpgamgr0>;
 
@@ -531,6 +531,7 @@
 			reg = <0xff400000 0x100000>;
 			resets = <&rst LWHPS2FPGA_RESET>;
 			clocks = <&l4_main_clk>;
+			status = "disabled";
 		};
 
 		fpga_bridge1: fpga_bridge@ff500000 {
@@ -538,6 +539,21 @@
 			reg = <0xff500000 0x10000>;
 			resets = <&rst HPS2FPGA_RESET>;
 			clocks = <&l4_main_clk>;
+			status = "disabled";
+		};
+
+		fpga_bridge2: fpga-bridge@ff600000 {
+			compatible = "altr,socfpga-fpga2hps-bridge";
+			reg = <0xff600000 0x100000>;
+			resets = <&rst FPGA2HPS_RESET>;
+			clocks = <&l4_main_clk>;
+			status = "disabled";
+		};
+
+		fpga_bridge3: fpga-bridge@ffc25080 {
+			compatible = "altr,socfpga-fpga2sdram-bridge";
+			reg = <0xffc25080 0x4>;
+			status = "disabled";
 		};
 
 		fpgamgr0: fpgamgr@ff706000 {
@@ -766,7 +782,7 @@
 		};
 
 		qspi: spi@ff705000 {
-			compatible = "cdns,qspi-nor";
+			compatible = "intel,socfpga-qspi", "cdns,qspi-nor";
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0xff705000 0x1000>,
@@ -813,6 +829,7 @@
 			num-cs = <4>;
 			clocks = <&spi_m_clk>;
 			resets = <&rst SPIM0_RESET>;
+			reset-names = "spi";
 			status = "disabled";
 		};
 
@@ -825,6 +842,7 @@
 			num-cs = <4>;
 			clocks = <&spi_m_clk>;
 			resets = <&rst SPIM1_RESET>;
+			reset-names = "spi";
 			status = "disabled";
 		};
 
diff --git a/arch/arm/boot/dts/socfpga_cyclone5_de10_nano.dts b/arch/arm/boot/dts/socfpga_cyclone5_de10_nano.dts
new file mode 100644
index 000000000000..21a651df51df
--- /dev/null
+++ b/arch/arm/boot/dts/socfpga_cyclone5_de10_nano.dts
@@ -0,0 +1,198 @@
+/// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2017, Intel Corporation
+ *
+ * based on socfpga_cyclone5_de0_nano_soc.dts
+ *
+ * VincNL test
+ */
+
+#include "socfpga_cyclone5.dtsi"
+
+/ {
+        model = "Terasic DE10-Nano";
+        compatible = "altr,socfpga-cyclone5", "altr,socfpga";
+
+        chosen {
+                bootargs = "earlyprintk";
+                stdout-path = "serial0:115200n8";
+        };
+
+        aliases {
+                ethernet0 = &gmac1;
+        };
+
+        memory {
+                name = "memory";
+                device_type = "memory";
+                reg = <0x0 0x40000000>; /* 1GB */
+        };
+
+        leds {
+               compatible = "gpio-leds";
+               hps0 {
+                       label = "hps_led0";
+                       gpios = <&portb 24 0>;
+                       linux,default-trigger = "heartbeat";
+               };
+        };
+
+        soc {
+                led_pio: gpio@0xff203000 {
+                        compatible = "altr,pio-1.0";
+                        reg = <0xff203000 0x10>;
+                        altr,gpio-bank-width = <7>;
+                        altr,ngpio = <7>;
+                        resetvalue = <0>;
+                        #gpio-cells = <2>;
+                        gpio-controller;
+                };
+
+                dipsw_pio: gpio@0xff204000 {
+                        compatible = "altr,pio-1.0";
+                        reg = <0xff204000 0x00000010>;
+                        interrupt-parent = <&intc>;
+                        interrupts = <0 40 1>;
+                        altr,gpio-bank-width = <4>;
+			altr,ngpio = <4>;
+                        altr,interrupt-type = <3>;
+                        altr,interrupt_type = <3>;
+                        edge_type = <2>;
+                        level_trigger = <0>;
+                        resetvalue = <0>;
+                        #gpio-cells = <2>;
+                        gpio-controller;
+                };
+
+
+                button_pio: gpio@0xff205000 {
+                        compatible = "altr,pio-1.0";
+                        reg = <0xff205000 0x00000010>;
+                        interrupt-parent = <&intc>;
+                        interrupts = <0 41 1>;
+                        altr,gpio-bank-width = <2>;
+			altr,ngpio = <2>;
+                        altr,interrupt-type = <2>;
+                        altr,interrupt_type = <2>;
+                        edge_type = <1>;
+                        level_trigger = <0>;
+                        resetvalue = <0>;
+                        #gpio-cells = <2>;
+                        gpio-controller;
+                };
+
+
+                h2f_pio: gpio@0xff206000 {
+                        compatible = "altr,pio-1.0";
+                        reg = <0xff206000 0x10>;
+                        altr,gpio-bank-width = <32>;
+                        altr,ngpio = <32>;
+                        resetvalue = <0>;
+                        #gpio-cells = <2>;
+                        gpio-controller;
+                };
+
+
+                f2h_pio: gpio@0xff207000 {
+                        compatible = "altr,pio-1.0";
+                        reg = <0xff207000 0x00000010>;
+                        interrupt-parent = <&intc>;
+                        interrupts = <0 43 1>;
+                        altr,gpio-bank-width = <32>;
+                        altr,interrupt-type = <3>;
+                        altr,interrupt_type = <3>;
+                        edge_type = <2>;
+                        level_trigger = <0>;
+                        resetvalue = <0>;
+                        #gpio-cells = <2>;
+                        gpio-controller;
+                };
+		
+		dp_0_frame_buf: vip@0xff200000 {
+			compatible = "altr,vip-frame-buffer-2.0";
+			reg = <0xff200000 0x40>;
+			interrupts = <0 44 1>;
+			altr,interrupt-type = <1>;
+			altr,max-width = <640>;
+			altr,max-height = <480>;
+			altr,bits-per-symbol = <8>;
+			altr,mem-port-width = <128>;
+		};
+        };
+
+};
+
+&gmac1 {
+        status = "okay";
+        phy-mode = "rgmii";
+
+        rxd0-skew-ps = <420>;
+        rxd1-skew-ps = <420>;
+        rxd2-skew-ps = <420>;
+        rxd3-skew-ps = <420>;
+        txen-skew-ps = <0>;
+        txc-skew-ps = <1860>;
+        rxdv-skew-ps = <420>;
+        rxc-skew-ps = <1680>;
+};
+
+&gpio0 {
+        status = "okay";
+};
+
+&gpio1 {
+        status = "okay";
+};
+
+&gpio2 {
+        status = "okay";
+};
+
+&porta {
+        bank-name = "porta";
+};
+
+&portb {
+        bank-name = "portb";
+};
+
+&portc {
+        bank-name = "portc";
+};
+
+&mmc0 {
+        status = "okay";
+};
+
+&usb1 {
+        status = "okay";
+};
+
+&uart0 {
+        status = "okay";
+};
+
+&watchdog0 {
+        status = "disabled";
+};
+
+&fpga_bridge0 {
+        status = "okay";
+        bridge-enable = <1>;
+};
+
+&fpga_bridge1 {
+        status = "okay";
+        bridge-enable = <1>;
+};
+
+&fpga_bridge2 {
+        status = "okay";
+        bridge-enable = <1>;
+};
+
+&fpga_bridge3 {
+        status = "okay";
+        bridge-enable = <1>;
+};
+
-- 
2.35.1

